working/augmentation-libraries/feature-specification.md

```diff
@@ -938,79 +728,14 @@ It is a compile-time error if:
-
-*   The introductory constructor has a super initializer _(super
-    constructor invocation at the end of the initializer list)_ and the
-    augmenting constructor does too. _An augmentation can replace the
-    implicit default `super()` with a concrete super-invocation, but cannot
-    replace a declared super constructor._ **(TODO: Why not? We allow
-    "replacing implementation", and this is *something* like that.)**
-
-*   The resulting constructor is not valid *(it has a redirection as well as
-    some initializer list elements, or it has multiple `super` initializers,
-    etc)*.
-
-*   A non-redirecting constructor augments a constructor which is not
-    potentially non-redirecting.
-
-*   A redirecting constructor augments a constructor which is not potentially
-    redirecting.
-
-#### Non-redirecting generative constructors
-
-These are probably the most complex constructors, but also the most common.
-
-At a high level, a non-redirecting generative constructor marked `augment` may:
-
-*   Augment the constructor with an _additional_ constructor body (bodies are
-    invoked in augmentation order, starting at the introductory declaration).
-
-*   Add initializers (and/or asserts) to the initializer list, as well as a
-    `super` call at the end of the initializer list.
-
-#### Non-redirecting factory constructors
-
-A non-redirecting factory constructor marked `augment` works in the same way as
-a normal function augmentation.
-
-If it has a body, it replaces the body of the augmented constructor (if
-present), and it may invoke the augmented body by calling
-`augmented(arguments)`.
-
-#### Redirecting generative constructors
-
-A redirecting generative constructor marked `augment` adds its redirection
-to the augmented constructor.
-
-This converts it into a redirecting generative constructor, removing the
-potentially non-redirecting property of the constructor.
-
-It is a compile-time error if:
-
-*   The augmented constructor has any initializers.
-*   The augmented constructor has a body.
-*   The augmented constructor has a redirection.
-
-This redirecting generative constructor now behaves exactly like any other
-redirecting generative constructor when it is invoked.
-
-#### Redirecting factory constructors
-
-A redirecting factory constructor marked `augment` adds its factory redirection
-*(e.g., `= C<int>.name`)* to the augmented constructor.
-
-The result of applying the augmenting constructor is a redirecting factory
-constructor with the same target constructor designation as the augmenting
-constructor. This removes the potentially non-redirecting property of the
-constructor.
-
-It is a compile-time error if:
-
-*   The augmented factory constructor has a body, or it is redirecting.
+    constructor is not, or vice versa. *An augmentation can't change whether or
+    not a constructor is generative because that affects whether users are
+    allowed to call the constructor in a subclass's initializer list.*
```

Comment:
(That only argues for why *adding* `factory` is bad, not why removing it is.)

Reply:
An augmentation can't really *remove* anything. If we did allow:

```dart
class C {
  factory C();
}

augment class C {
  augment C() { return ...; }
}
```

Then users would probably reasonably assume that the resulting class took the union of the modifiers like it does with `implements`, `with`, etc.